lib/crypto/random.js

```diff
@@ -4,4 +4,32 @@
 
 'use strict'
 
-module.exports = require('../promise').promisify(require('crypto').randomBytes)
+const randomBytes = require('../promise').promisify(require('crypto').randomBytes)
+
+function random(bytes) {
+  if (Array.isArray(bytes)) {
+    const sum = bytes.reduce((acc, val) => acc + val, 0)
+    return randomBytes(sum).then(buf => {
+      let pos = 0
+      return bytes.map(num => {
+        const slice = buf.slice(pos, pos + num)
+        pos += num
+        return slice
+      })
+    })
+  } else {
+    return randomBytes(bytes)
+  }
+}
+
+random.hex = bytes => {
+  return random(bytes).then(bufs => {
+    if (! Array.isArray(bufs)) {
```

Comment:
Another one from the ignore-if-you-disagree camp, but I generally think code reads a bit easier if conditions are swapped to eliminate `!` where possible.